docs/src/main/asciidoc/kafka-streams.adoc

```diff
@@ -722,3 +722,226 @@ For more details about the health information, see the
+                                                        .productId(),
+                                                (customerOrder, product) -> {
+                                                    EnrichedOrder enrichedOrder = new EnrichedOrder();
+                                                    enrichedOrder.setProduct(product);
+                                                    enrichedOrder.setCustomer(customerOrder.customer);
+                                                    enrichedOrder.setOrder(customerOrder.order);
+                                                    return enrichedOrder;
+                                                })
+                        )
+                )
+        );
+    }
+----
+
+Here we have 3 inputs. The first function takes a `KStream` and its output is another `Function` that takes a `GlobalKTable` as its input and another function as its output.
+This last function takes another `GlobalKTable` as its input and a `KStream` is provided as this function's output which will be used as the processor's output.
+In this example, we have three curried functions. Behind the scenes, the binder will call the `apply` method on those functions in the order that they appear. 
+
+Here is the corresponding binding interface for this application.
+
+[source]
+----
+interface CustomGlobalKTableProcessor {
+
+        @Input("input-1")
+        KStream<?, ?> input1();
+
+        @Input("input-2")
+        GlobalKTable<?, ?> input2();
+
+        @Input("input-3")
+        GlobalKTable<?, ?> input3();
+
+        @Output("output")
+        KStream<?, ?> output();
+}
+----
+
+Here also, the input bindings follow a natural order.
+
+==== Multiple functions in the same application
+
+Multiple functions aan be defined in the same application.
+When doing this, the binder will do a natural sorting on multiple function bean names first and then apply input and output bindings on them in the natural order.
+Consider the following two function beans in the same application.
+
+[source]
+----
+    @Bean
+    public Function<KStream<?, String>, KStream<?, WordCount>> process1() {
+
+    }
+
+    @Bean
+    public Function<KStream<?, String>, KStream<?, WordCount>> process2() {
+
+    }
+----
+
+Consider also the following binding interface.
+
+
+[source]
+----
+interface Bindings {
+
+    @Input("input-1")
+    KStream<?, ?> input1();
+
+    @Input("input-2")
+    KStream<?, ?> input2();
+
+    @Ouput("output-1")
+    KStream<?, ?> output1();
+
+    @Output("output-2")
+    KStream<?, ?> output1();
+}
+----
+
+Binder will first take the method `process1` and use input binding `input-1` and output binding `output-1`.
+Similarly, for the method `process2`, it will use input binding `input-2` and output binding `output-2`.
+
+==== Using custom state stores in functional applications
+
+You can define custom state stores as beans in your application and those will be detected and added to the Streams builder by the binder.
```

Comment:
Is "Streams builder" KStreams builder? If yes, we could make it explicit to avoid a generic terminology here - it could confuse folks.